src/app/users/get-user.ts

```diff
@@ -1,65 +1,99 @@
     "app.getUserForLogin",
     { [SemanticAttributes.CODE_FUNCTION]: "getUserForLogin" },
     async () => {
-      const user = await users.findById(userId as UserId)
+      const user = await users.findById(userId)
+
       if (user instanceof Error) {
         return user
       }
-      addAttributesToCurrentSpan({
-        [ENDUSER_ALIAS]: user.username,
-      })
-      user.lastConnection = new Date()
 
-      // IP tracking logic could be extracted into domain
+      // this routing run asynchrously, to update metadata on the background
+      updateUserIpInfo({ userId, ip, logger })
+
+      return user
+    },
+  )
+
+const updateUserIpInfo = async ({
+  userId,
+  ip,
+  logger,
+}: {
+  userId: UserId
+  ip?: Ip
+  logger: Logger
+}): Promise<void> =>
+  asyncRunInSpan(
+    "app.addIp",
+    { [SemanticAttributes.CODE_FUNCTION]: "addIp" },
+    async () => {
       const ipConfig = getIpConfig()
-      if (ip && ipConfig.ipRecordingEnabled) {
-        const lastIP: IPType | undefined = user.lastIPs.find(
-          (ipObject: IPType) => ipObject.ip === ip,
-        )
-        if (lastIP) {
-          lastIP.lastConnection = user.lastConnection
-        } else if (ipConfig.proxyCheckingEnabled) {
+
+      const lastConnection = new Date()
+
+      const userIp = await usersIp.findById(userId)
+
+      if (userIp instanceof RepositoryError) throw userIp
+
+      if (!ip || !ipConfig.ipRecordingEnabled) {
+        const result = await usersIp.update(userId, lastConnection)
+
+        if (result instanceof Error) {
+          logger.error(
+            { result, userId, ip },
+            "impossible to update user last connection",
+          )
+        }
+
+        return
+      }
+
+      const lastIP = userIp.lastIPs.find((ipObject) => ipObject.ip === ip)
+
+      if (lastIP) {
+        lastIP.lastConnection = lastConnection
+      } else {
+        let ipInfo = {
+          ip,
+          firstConnection: lastConnection,
+          lastConnection: lastConnection,
+        } as IPType
+
+        if (ipConfig.proxyCheckingEnabled) {
           const ipFetcher = IpFetcher()
-          const ipInfo = await ipFetcher.fetchIPInfo(ip as IpAddress)
-          if (!(ipInfo instanceof Error)) {
-            user.lastIPs.push({
-              ip,
-              ...ipInfo,
-              Type: ipInfo.type,
-              firstConnection: user.lastConnection,
-              lastConnection: user.lastConnection,
-            })
+          const ipFetcherInfo = await ipFetcher.fetchIPInfo(ip as IpAddress)
+
+          if (!(ipFetcherInfo instanceof Error)) {
+            ipInfo = { ...ipInfo, ...ipFetcherInfo }
           }
         }
+        userIp.lastIPs.push(ipInfo)
       }
+      const result = await usersIp.update(userId, lastConnection, userIp.lastIPs)
 
-      const updateResult = await users.update(user)
-
-      if (updateResult instanceof RepositoryError) {
-        return updateResult
+      if (result instanceof Error) {
```

Comment:
Again don't eat the error. Always return it - it will automatically be reported via OpenTelemetry.